tests/lax_numpy_test.py

```diff
@@ -3905,5 +3906,49 @@ def f(x):
     check_grads(f, (1.,), order=1)
 
 
+  def testWrappedSignaturesMatch(self):
+    """Test that jax.numpy function signatures match numpy."""
+    # Note: testing signatures for an exact match is problematic, because numpy's signatures
+    # evolve from release to release (mainly adding new parameters when necessary), and jax.numpy
+    # adds additional jax-specific parameters to some functions. Because of this, the test checks
+    # that jax.numpy function parameters are a superset of the wrapped numpy function parameters.
+    # That way, when tested against older numpy versions (where functions often have fewer parameters),
+    # the test should pass, and when run against new numpy releases, the tests will fail indicating
+    # where jax functionality must be upgraded to match.
+
+    # TODO(jakevdp): fix the following signatures.
+    skip = {
+      'allclose', 'amax', 'amin', 'angle', 'argmax', 'argmin', 'around', 'broadcast_to',
+      'clip', 'convolve', 'corrcoef', 'correlate', 'cumprod', 'cumproduct', 'cumsum',
+      'diff', 'empty_like', 'einsum', 'einsum_path',
+      'eye', 'full', 'full_like', 'gradient', 'histogram',
+      'isposinf', 'isneginf', 'isscalar', 'max', 'min', 'nancumprod', 'nancumsum',
+      'nanprod', 'nansum', 'ones', 'ones_like', 'pad', 'polyadd', 'polyder', 'polysub',
+      'prod', 'product', 'round', 'stack', 'sum', 'tile', 'zeros_like'
+    }
+    jnp_funcs = {name: getattr(jnp, name) for name in dir(jnp) if name not in skip}
+    func_pairs = {name: (fun, fun.__np_wrapped__) for name, fun in jnp_funcs.items()
+                  if hasattr(fun, '__np_wrapped__')}
+    assert len(func_pairs) > 0
+
+    mismatches = {}
+
+    for name, (jnp_fun, np_fun) in func_pairs.items():
+      # Note: can't use inspect.getfullargspec due to numpy issue
+      # https://github.com/numpy/numpy/issues/12225
+      try:
+        np_params = list(inspect.signature(np_fun).parameters)
+      except ValueError:
+        # Some functions cannot be inspected
+        continue
+      jnp_params = list(inspect.signature(jnp_fun).parameters)
+      if set(jnp_params).issubset({'args', 'kwargs'}):
```

Comment:
Rather than relying on magic names for `*args` and `**kwargs`, it seems like it would be more reliable to look at [`Parameter.kind`](https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind)?

Reply:
Done.